### ISSUE TITLE
Cannot manage launchdarkly_feature_flag_environment with rules containing boolean clauses

### ISSUE THREAD

**Issue by mvisonneau** [launchdarkly/terraform-provider-launchdarkly#51]
Assuming the following existing configuration:

![image](https://user-images.githubusercontent.com/1761583/110624635-61c04480-8196-11eb-91ee-b21e3a18093f.png)

Importing it into the statefile or attempting to perform plan upon this resource results into the following error:

```
Error: failed to set rules on flag with key "dark-mode": rules.0.clauses.0.values.0: '' expected type 'string', got unconvertible type 'bool'
```

**Comment by mvisonneau** [comment]
Thanks, I assume it has something to do with: https://github.com/launchdarkly/terraform-provider-launchdarkly/blob/master/launchdarkly/clause_helper.go#L26:L26

**Comment by ldhenry** [comment]
Yeah, you got it. In order to avoid introducing a breaking change we will likely add a `"valueType"` argument to the clause schema with a default of `"string"` that will let you specify what type of values the provider should use for the `"values"` array. 

Unfortunately this means that we will not be able to support polymorphic types in the values array in a v1.X release, but I think this is acceptable for the majority of use cases.  

For v2 we will likely break the schema to support polymorphic value types.

**Comment by ldhenry** [comment]
Hi @mvisonneau,

I just published v1.5.1 of the provider which includes a fix for this issue. After upgrading, you will have to set the [rule clause's `value_type`](https://registry.terraform.io/providers/launchdarkly/launchdarkly/latest/docs/resources/feature_flag_environment#nested-clauses-blocks) to `boolean` in order to manage rules with boolean clauses.

Thanks for reporting this,
Henry

**Comment by mvisonneau** [comment]
Brilliant, thanks a lot @ldhenry, it works like a charm now :bow: :ok_hand:
